test/other.jl

```diff
@@ -2,6 +2,34 @@ using Metalhead, Test
 using Flux
 
 @testset "MLPMixer" begin
-    @test size(MLPMixer()(rand(Float32, 256, 256, 3, 2))) == (1000, 2)
-    @test_skip gradtest(MLPMixer(), rand(Float32, 256, 256, 3, 2))
-end
\ No newline at end of file
+	@testset for mode in [:small, :base, :large, :huge]
+		@testset for drop_path_rate in [0.0, 0.5, 0.99]
+			m = MLPMixer(mode; drop_path_rate)
+			@test size(m(rand(Float32, 224, 224, 3, 2))) == (1000, 2)
+			@test_skip gradtest(m, rand(Float32, 224, 224, 3, 2))
+			GC.gc()
+		end
+	end
+end
+
+@testset "ResMLP" begin
+	@testset for mode in [:small, :base, :large, :huge]
+		@testset for drop_path_rate in [0.0, 0.5, 0.99]
+			m = ResMLP(mode; drop_path_rate)
+			@test size(m(rand(Float32, 224, 224, 3, 2))) == (1000, 2)
+			@test_skip gradtest(m, rand(Float32, 224, 224, 3, 1))
+			GC.gc()
+		end
+	end
+end
+
+@testset "gMLP" begin
+	@testset for mode in [:small, :base, :large, :huge]
+		@testset for drop_path_rate in [0.0, 0.5, 0.99]
+			m = gMLP(mode; drop_path_rate)
+			@test size(m(rand(Float32, 224, 224, 3, 2))) == (1000, 2)
+			@test_skip gradtest(m, rand(Float32, 224, 224, 3, 2))
+			GC.gc()
```

Comment:
Regarding OOMs, it seems there are a number of models which can accommodate smaller image sizes. I think we can get away with CIFAR-10's 32x32 for those. ImageNet size can be reserved for the smaller variants of each model type if need be, just to make sure the overall architecture is sound.

Reply:
That messes up the config values for the depth sometimes, since CIFAR is a smaller dataset and sizes may not match - they will for some of the models where the sizes are held constant throughout, thought, I get that. My idea was basically to make sure the config dicts hold up the way they should - we could always shift to a smaller image size wherever we can. But `gradtests` are not even enabled and there are OOMs, so that's definitely a bit of a problem in case those have to be enabled in the future